src/server/utils/wallets/createAwsKmsWallet.ts

```diff
@@ -1,25 +1,42 @@
 import { CreateKeyCommand, KMSClient } from "@aws-sdk/client-kms";
-import { WalletType } from "../../../schema/wallet";
-import { getConfig } from "../../../utils/cache/getConfig";
+import {
+  FetchAwsKmsWalletParamsError,
+  fetchAwsKmsWalletParams,
+  type AwsKmsWalletParams,
+} from "./fetchAwsKmsWalletParams";
 import { importAwsKmsWallet } from "./importAwsKmsWallet";
 
-interface CreateAwsKmsWalletParams {
+type CreateAwsKmsWalletParams = {
   label?: string;
-}
+} & Partial<AwsKmsWalletParams>;
 
+export class CreateAwsKmsWalletError extends Error {}
+
+/**
+ * Create an AWS KMS wallet, and store it into the database
+ * All optional parameters are overrides for the configuration in the database
+ * If any required parameter cannot be resolved from either the configuration or the overrides, an error is thrown.
+ * If credentials (awsAccessKeyId and awsSecretAccessKey) are explicitly provided, they will be stored separately from the global configuration
+ */
 export const createAwsKmsWallet = async ({
   label,
+  ...overrides
 }: CreateAwsKmsWalletParams): Promise<string> => {
-  const config = await getConfig();
-  if (config.walletConfiguration.type !== WalletType.awsKms) {
-    throw new Error(`Server was not configured for AWS KMS wallet creation`);
+  let kmsWalletParams: AwsKmsWalletParams;
+  try {
+    kmsWalletParams = await fetchAwsKmsWalletParams(overrides);
+  } catch (e) {
+    if (e instanceof FetchAwsKmsWalletParamsError) {
+      throw new CreateAwsKmsWalletError(e.message);
+    }
+    throw e;
   }
```

Comment:
What's the purpose of this try/catch + custom error type, vs just letting the code throw?
```typescript
const kmsWalletParams = await fetchAwsKmsWalletParams(overrides);
// This may throw to the caller
```

Reply:
create backend wallet endpoint uses the `createAwsKmsWallet` method. By having a custom error class it can then know for sure error is 400

Reply:
the full version of this is to have an EngineServiceError that also contains more metadata, and each service can extend this. That would be a new PR though